### DISCUSSION TITLE
How to add a textbox where some text can be typed in the TUI

### DISCUSSION THREAD

**Discussion by ShafaqFatimaMughal** [Textualize/textual#945] [General]
I am trying to build some kind of a TUI that serves as a todo list. For this, I need to use Textual to create some kind of a textbox where a user can type a todo in the terminal. Most of the examples (that I have seen) in the repository focus on the mouse. That is why I have not been able to figure out.

**Comment by davep** [comment]
Hi! If you're currently working with the current main branch (0.1.18) there isn't a text input. Coming next Monday (2022-10-24) is a brand new release of Textual, with lots of additions; one of these is a text input widget.
